src/app/feature/map/search/search.component.ts

```diff
@@ -39,78 +26,74 @@ export class SearchComponent implements OnInit {
   constructor(
     private readonly _mapService: MapService,
     private readonly _mapDataProviderService: MapDataProviderService,
-    private readonly fb: FormBuilder,
     private readonly translateService: TranslateService
   ) {
-    this.stateGroups = [];
+    this.destinationGroups = [];
+    this.userLocationGroups = [];
   }
 
-  @Output() public setLocation = new EventEmitter<LocationNode>();
-
-  public locations: GuideMapFeaturePoint[];
-
-  public stateForm: FormGroup = this.fb.group({
-    stateGroup: '338',
-  });
+  @Output() set location(value: string) {
+    value && value.length && this.findUserLocation(value);
+  }
+  @Output() set destination(value: string) {
+    value && value.length && this.findDestination(value);
+  }
 
-  public stateGroups: StateGroup[];
+  public featurePoints: GuideMapFeaturePoint[];
 
-  public stateGroupOptions: Observable<StateGroup[]>;
+  public destinationGroups: IOptionGroup[];
+  public userLocationGroups: IOptionGroup[];
 
   public ngOnInit(): void {
     this._mapDataProviderService.getFeaturePoints().subscribe((data) => {
```

Comment:
We need to discuss it
Please don't use `getFeaturePoints` every time, we call this only init method from map
Use `qrCodes` getter for it
And check `_initFeaturePoints` method logic